core/src/main/java/org/apache/iceberg/inmemory/InMemoryCatalog.java

```diff
@@ -288,6 +288,7 @@ public List<Namespace> listNamespaces(Namespace namespace) throws NoSuchNamespac
 
     List<Namespace> filteredNamespaces =
         namespaces.keySet().stream()
+            .filter(n -> !n.isEmpty())
```

Comment:
this would previously fail in L303 when calling `listNamespaces(Namespace.empty())`

Reply:
So does this one support empty namespaces or not?

Reply:
Just wondering what the behavior should be here, since it is a valid namespace so shouldn't we be able to list it?

Reply:
yes, `InMemoryCatalog` supports empty namespaces and this comes back to the TODO I added in https://github.com/apache/iceberg/pull/9890/files#diff-de393039ffcf049ebb22892ac1b9aad88d53274c1e86431b45e776e4568246f8R1037 to determine what the listing behavior with empty namespaces should be